specs/index.rst

```diff
@@ -42,105 +52,145 @@ asset path
 .. note:: All relative paths **MUST** be relative to the folder containing the *assets-manifest* file.
 
 
-File formats
-------------
+An example
+""""""""""
 
-Version 1.0
-"""""""""""
+A pipeline is generating all the CSS and Javascript required for *example.com*. There are four bundles:
 
-An *assets-manifest* file is a JSON file, usually named ``assets-manifest.json``.
-Its toplevel entry **MUST** be a dictionary.
+ * site-wide CSS (logical path ``site.css``)
+ * site-wide Javascript (logical path ``site.js``)
+ * Javascript for a subscription flow, including additional code for communicating with a payment provider (logical path ``subscribe.js``); this is used with ``site.js`` rather than instead of it
+ * CSS for a "lightbox" media browser (logical path ``lightbox.css``); this is used instead of ``site.css``
 
-Example:
+In production, the pipeline will render each bundle as a single asset, including a hash-based fingerprint in the asset path. For instance, the bundle identified by the logical path ``site.css`` might be rendered as an asset with the asset path ``site-4fbcc857.css``; the bundle ``site.js`` rendered as ``site-14ffdec5.js`` and so forth.
 
-.. code-block:: json
+Additionally, each bundle may also have a sourcemap rendered alongside it; the sourcemap for the ``site.js`` bundle would be a bundle with logical path ``site.js.map``, and might result in an asset with asset path ``site-14ffdec5.js.map``
+
+.. warning::
+
+   Sourcemaps **SHOULD** be listed in the manifest, under a logical
+   path created by concatenating ".map" to the end of the logical path
+   for the relevant bundle.
+
+.. note::
+
+   Although it is helpful for any fingerprint used in the asset path
+   of a sourcemap to match the fingerprint of its related rendered
+   asset, this is not always practical depending on how a particular
+   pipeline is configured.
```

Comment:
This is exactly why people find sourcemaps hard :-)

If the URL to the sourcemap is precisely the URL to the asset but with ".map" on the end, then the browser will find it automatically, without needing a `sourceMappingUrl` comment. How about:

```rst
.. note::

    For sourcemaps to be automatically discovered, the fingerprint in its asset path
    should match the fingerprint of the rendered asset that the sourcemap is related to:

     * if the ``site.js`` bundle has an asset ``site.45bf76ed.js``
     * then the ``site.js.map`` bundle has an asset ``site.45bf76ed.js.map``

    Otherwise, an explicit linking comment will be required in the rendered asset:

     * if the ``site.js`` bundle has an asset ``site.45bf76ed.js``
     * and the ``site.js.map`` bundle has an asset ``site.1455d2ef.js.map``
     * then the asset ``site.45bf76ed.js`` should contain the comment
       ``//# sourceMappingUrl=site.1455d2ef.js.map``
```

(You're also making assumptions about how people generate fingerprints with dependent assets; most tools get this wrong, and while it might be useful at some point to issue a separate guidance note, I'd argue that this isn't the place to do it.)